### PR TITLE
backport some changes from https://github.com/arangodb/arangodb/pull/13507

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

This is a preparation PR for backporting some further replication tests and bugfixes from devel to 3.7. 
It isolates some of the changes done in https://github.com/arangodb/arangodb/pull/13507 into a small PR to facilitate reviewing. 
A further PR with the added replication tests and a bugfix for 3.7 will follow up.

This PR extends the length of some internal byte buffers from 11 to 21, so that they cannot overflow, regardless of what input is coming in. The overflow was theoretically possible before, but only with manipulated timestamp input.

The PR also fixes an issue in a RocksDB heap helper class that does a potentially unsafe self-move assignment. This issue was raised when running tests with a debug STL build (`_GLIBCXX_DEBUG`).

- [x] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [x] :book: CHANGELOG entry made

#### Backports:

- [x] No backports required

### Testing & Verification

*(Please pick either of the following options)*

- [x] This change is a trivial rework / code cleanup without any test coverage.
- [x] The behavior in this PR was *manually tested*
- [x] I ensured this code runs with ASan / TSan or other static verification tools (`_GLIBCXX_DEBUG`)

Link to Jenkins PR run:
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13912/

### CODE REVIEW

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13918/

File: CHANGELOG [hunk 1]
```diff
@@ -1,6 +1,12 @@
 v3.7.8 (XXXX-XX-XX)
 -------------------
 
+* Fixed a RocksDB bug which could lead to an assertion failure when compiling
+  with STL debug mode -D_GLIBCXX_DEBUG.
+
+* Fixed a rare internal buffer overflow around ridBuffers.
+
```
Comment (jsteemann) [inline CHANGELOG:8]:
```suggestion
* Fixed a rare internal buffer overflow around ridBuffers.
```

Comment (jsteemann) [inline CHANGELOG:8]:
```suggestion
* Fixed a rare internal buffer overflow around ridBuffers.
```

File: CHANGELOG [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
+
```
Comment (jsteemann) [inline CHANGELOG:9]:
```suggestion
* Fixed a rare internal buffer overflow around ridBuffers.

```

**Review by neunhoef — APPROVED** [review]
LGTM

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13925/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13944/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13949/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13953/

**Comment by jsteemann** [comment]
Tests blue

**Pull request merged**